chrome/content/zotfile/zotfile.js

```diff
@@ -455,16 +455,15 @@ Zotero.ZotFile = new function() {
      */
     this.formatSubfolder = function(item, format) {
         if (format == '') return '';
-        if(format.slice(-1) == this.folderSep) format = format.slice(0, -1);
+        if (format.slice(-1) == this.folderSep) format = format.slice(0, -1);
+        if (format[0] == this.folderSep) format = format.slice(1);
         var subfolder = this.Wildcards.replaceWildcard(item, format);
-        if (subfolder[0] == this.folderSep) subfolder = subfolder.slice(1);
-        // replace invalid characters        
+        // replace invalid characters
         subfolder = OS.Path.split(subfolder).components
             .map(s => s == '' ? 'undefined' : s)
+            .filter(s => s != this.Wildcards.emptyCollectionPlaceholder)
             .map(s => Zotero.File.getValidFileName(s))
             .join(this.folderSep);
-        if (format.replace(/[\/\\]/g,'') == '%c')
-            subfolder = subfolder.replace(/undefined[\/\\]?/g, '');
```

Comment:
This is introduced in https://github.com/jlegewie/zotfile/commit/037be4cb681f31da625d1152d007b051e5fb5366 . This part can already be covered.